class-two-factor-core.php

```diff
@@ -204,6 +204,9 @@ public static function get_providers_registered() {
 		/**
 		 * Filter the supplied providers.
 		 *
+		 * This lets third-parties either remove providers (such as Email), or
```

Comment:
You can't remove providers here since the default ones will always be merged in line 217, right? This will likely break implementations like https://github.com/WordPress/wporg-two-factor/blob/fe0a981432ecd52f6b74a31cd2c7d26dd761e3df/wporg-two-factor.php#L81-L93

Reply:
Indeed.

This `get_providers_registered()` is intended mostly for the uninstall logic so it is attempting to collect every possible provider that could have been registered at any point, or any overrides.

I'll think about this some more.

Reply:
This is being fixed in #651.